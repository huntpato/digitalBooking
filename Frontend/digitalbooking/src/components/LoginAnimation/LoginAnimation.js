import React from 'react'
import styles from './LoginAnimation.module.css'
import logo from "../../assets/logo.png";

export const LoginAnimation = (props) => {
    const {animation}= styles; 
    const {message} = props;

    return (
        <div className= { animation }>
            <img src={ logo } alt="Digital Booking Logo"/>
            <h6>{ message }</h6>
        </div>
    )
}
