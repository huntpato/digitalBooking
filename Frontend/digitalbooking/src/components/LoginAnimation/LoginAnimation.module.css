.animation {
  align-items: center;
  background-color: rgba(255, 255, 255, 0.8);
  border-color: var(--color-typo2);
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.15);
  display: flex;
  flex-direction: column;
  justify-content: center;
  height: 100%;
  position: fixed;
  width: 100%;
  z-index: 5;
}

.animation img {
  -webkit-animation: heartbeat 5s ease-in-out infinite both;
  animation: heartbeat 5s ease-in-out infinite both;
}

.animation h6 {
  color: var(--color-typo2);
  font-size: 12px;
  padding-top: 20px;
}

@-webkit-keyframes heartbeat {
  from {
    -webkit-transform: scale(1);
    transform: scale(1);
    -webkit-transform-origin: center center;
    transform-origin: center center;
    -webkit-animation-timing-function: ease-out;
    animation-timing-function: ease-out;
  }
  10% {
    -webkit-transform: scale(0.91);
    transform: scale(0.91);
    -webkit-animation-timing-function: ease-in;
    animation-timing-function: ease-in;
  }
  17% {
    -webkit-transform: scale(0.98);
    transform: scale(0.98);
    -webkit-animation-timing-function: ease-out;
    animation-timing-function: ease-out;
  }
  33% {
    -webkit-transform: scale(0.87);
    transform: scale(0.87);
    -webkit-animation-timing-function: ease-in;
    animation-timing-function: ease-in;
  }
  45% {
    -webkit-transform: scale(1);
    transform: scale(1);
    -webkit-animation-timing-function: ease-out;
    animation-timing-function: ease-out;
  }
}
@keyframes heartbeat {
  from {
    -webkit-transform: scale(1);
    transform: scale(1);
    -webkit-transform-origin: center center;
    transform-origin: center center;
    -webkit-animation-timing-function: ease-out;
    animation-timing-function: ease-out;
  }
  10% {
    -webkit-transform: scale(0.91);
    transform: scale(0.91);
    -webkit-animation-timing-function: ease-in;
    animation-timing-function: ease-in;
  }
  17% {
    -webkit-transform: scale(0.98);
    transform: scale(0.98);
    -webkit-animation-timing-function: ease-out;
    animation-timing-function: ease-out;
  }
  33% {
    -webkit-transform: scale(0.87);
    transform: scale(0.87);
    -webkit-animation-timing-function: ease-in;
    animation-timing-function: ease-in;
  }
  45% {
    -webkit-transform: scale(1);
    transform: scale(1);
    -webkit-animation-timing-function: ease-out;
    animation-timing-function: ease-out;
  }
}

