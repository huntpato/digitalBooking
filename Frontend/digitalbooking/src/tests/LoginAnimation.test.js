import React from 'react';
import { shallow } from 'enzyme';
import { LoginAnimation } from '../components/LoginAnimation/LoginAnimation'
import '@testing-library/jest-dom'
import { expect } from '@jest/globals';

jest.mock('lottie-web')
describe('Pruebas en componente <LoginAnimation />', ()=> {
    const message="Prueba Animacion"
    const wrapper= shallow(<LoginAnimation message={message}/>);

    test('Comprobar que <Button2 /> renderiza correctamente.', () => {
    
        expect( wrapper ).toMatchSnapshot();
      
    })

    test('Verificar que <LoginAnimation /> muestra el texto enviado por props', () => {
    
        const LoginAnimationH6 = wrapper.find('h6').text();
        expect( LoginAnimationH6 ).toContain( message );
    }) 

    test('Verificar que el componente carga la clase de CSS correctamente', () => {

        const div = wrapper.find('div');
        expect(div.hasClass('animation')).toBe(true);
    })
    

})
