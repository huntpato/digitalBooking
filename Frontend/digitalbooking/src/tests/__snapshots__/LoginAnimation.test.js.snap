// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <LoginAnimation /> Comprobar que <Button2 /> renderiza correctamente. 1`] = `
<div
  className="animation"
>
  <img
    alt="Digital Booking Logo"
    src="logo.png"
  />
  <h6>
    Prueba Animacion
  </h6>
</div>
`;
